warehouse/query-core/src/main/java/datawave/query/iterator/filter/field/index/FieldIndexFilter.java

```diff
@@ -0,0 +1,137 @@
+    // evaluation defaults to true.
+    public boolean keep(String ingestType, String fieldName, Value value) {
+        // Check the value to see if there is any data to filter against
+        if (fieldIndexFilterMapByType != null && !fieldIndexFilterMapByType.isEmpty() && fieldIndexFilterNodes != null && !fieldIndexFilterNodes.isEmpty()
+                        && ingestType != null && value != null && value.get().length > 0) {
+            Multimap<String,String> fieldValuesMap = decodeValue(value).getFilterData().getFieldValueMapping();
+            
+            // make sure that this field mapping is present for the given ingest type
+            Collection<String> filterFields = null;
+            Multimap<String,String> fieldIndexFilterMap = fieldIndexFilterMapByType.get(ingestType);
+            if (fieldIndexFilterMap != null) {
+                filterFields = new HashSet<>(fieldIndexFilterMap.get(fieldName));
+                if (filterFields != null) {
+                    filterFields.retainAll(fieldValuesMap.keys());
+                    filterFields.retainAll(fieldIndexFilterNodes.keySet());
+                }
+            }
+            
+            if (filterFields != null && !filterFields.isEmpty()) {
+                DatawaveJexlContext datawaveJexlContext = new DatawaveJexlContext();
+                List<JexlNode> nodes = new ArrayList<>();
+                
+                // Iterate over the intersected set of fields, and get a value tuple for each of the configured types
+                for (String filterField : filterFields) {
+                    Set<ValueTuple> valueTuples = new HashSet<>();
+                    for (String dataType : typeMetadata.getTypeMetadata(filterField, ingestType))
+                        // @formatter:off
+                        valueTuples.addAll(
+                                fieldValuesMap.get(filterField)
+                                        .stream()
+                                        .map(fieldValue -> createValueTuple(filterField, dataType, fieldValue))
+                                        .filter(valueTuple -> valueTuple != null)
+                                        .collect(Collectors.toSet()));
+                        // @formatter:on
+                    if (!valueTuples.isEmpty()) {
+                        datawaveJexlContext.set(filterField, valueTuples);
+                        nodes.addAll(fieldIndexFilterNodes.get(filterField));
+                    }
+                }
+                
+                if (!nodes.isEmpty()) {
+                    // AND all of the JEXL nodes together, create a script, and run the evaluation
+                    JexlNode rootNode = JexlNodeFactory.createAndNode(nodes);
+                    
+                    // Evaluate the JexlContext against the Script
+                    Script script = engine.createScript(JexlStringBuildingVisitor.buildQuery(rootNode));
```

Comment:
This seem expensive.  Can we not simply loop through the fields and do a comparison directly?